modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java

```diff
@@ -688,8 +698,10 @@ private void updateEventInternal(final String mpId, Opt<Date> startDateTime,
       String agentId = extendedEventDto.getCaptureAgentId();
       Opt<String> seriesId = Opt.nul(record.getSnapshot().get().getMediaPackage().getSeries());
 
+      // Check for conflicting events
       // Check scheduling conficts in case a property relevant for conflicts has changed
-      if (captureAgentId.isSome() || startDateTime.isSome() || endDateTime.isSome()) {
+      if ((captureAgentId.isSome() || startDateTime.isSome() || endDateTime.isSome())
+            && (!isAdmin() || (isAdmin() && !allowConflict))) {
```

Comment:
Slight performance optimization:

`&& (!allowConflicts || !isAdmin()) 
`

is more efficient as the user's role needs not to be determined if conflicts are not allowed anyway which is the case in 99.99% of calls to this method.

We likely can save one or more trees (think of energy consumption) by avoiding the execution of that code ;-)

Reply:
👍 :D